pilot/cmd/istioctl/log.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"istio.io/istio/pkg/log"
+)
+
+type component struct {
+	Name      string
+	Container string
+	Label     string
+}
+
+var (
+	controlPlaneComponents = map[string]component{
+		"pilot":   {Name: "pilot", Container: "discovery", Label: "istio=pilot"},
+		"mixer":   {Name: "mixer", Container: "mixer", Label: "istio=mixer"},
+		"ingress": {Name: "ingress", Container: "", Label: "istio=ingress"},
+		"ca":      {Name: "ca", Container: "", Label: "istio=istio-ca"},
+	}
+
+	// TODO - allow retrieval of proxy logs when using istioctl logs proxy <podID> [-n namespace]
+	logCmd = &cobra.Command{
+		Use:   "logs <component> [container]",
+		Short: "Retrieves logs from an Istio component [kube only]",
+		Long: `
+Retrieves logs from an Istio component when running in Kubernetes
+	
+Supported components are:
+
+	[pilot mixer ingress ca]
```

Comment:
How do you envision this to be used? `kubectl logs <istio-pilot>.` should yield the same info right? I think the todo you have above (by pod ID) is probably the key element in this PR. 

In the interest of keeping this platform independent, may be have a --k8s flag, such that we can extend this to other platforms as well?

Reply:
So the idea was just to streamline what people can already get as they would have to look up the pod ID in order to get this in kube. I wanted to keep it smaller for now but can add it into this PR rather than a later one if you like.

There already is a platform flag in istioctl, but I'm not sure how we would do this with consul/eureka because they don't concern themselves with managing/accessing logs so it completely depends on how the user has implemented their stack.

Reply:
kubectl logs -n istio-system -l istio=pilot -c discovery